saleor/graphql/giftcard/mutations.py

```diff
@@ -132,7 +132,18 @@ def clean_input(cls, info: ResolveInfo, instance, data, **kwargs):
 
         # perform only when gift card is created
         if instance.pk is None:
-            cleaned_input["code"] = generate_promo_code()
+            code = cleaned_input.get("code")
+            if code and not is_available_promo_code(code):
+                raise ValidationError(
+                    {
+                        "code": ValidationError(
+                            "Promo code already exists.",
+                            code=GiftCardErrorCode.ALREADY_EXISTS,
```

Comment:
Has to be with `.value`, it causes mypy fail right now:
```suggestion
                            code=GiftCardErrorCode.ALREADY_EXISTS.value,
```